devicetypes/smartthings/samsung-smart-doorlock.src/samsung-smart-doorlock.groovy

```diff
@@ -0,0 +1,402 @@
+/**
+ *  Samsung Smart Doorlock
+ *
+ *  Copyright 2018 Samsung SDS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+import physicalgraph.zigbee.zcl.DataType
+
+
+metadata {
+	definition (name: "Samsung Smart Doorlock", namespace: "Samsung SDS", author: "kyun.park", mnmn: "SmartThings", vid: "generic-lock") {
```

Comment:
if you use the "Samsung SDS" namespace, this DTH needs to be in a folder that matches, not "smartthings"

Reply:
Is this regarding upload process of DTH into the SmartThings github?

Reply:
the path for the file you've added is

`devicetypes/smartthings/samsung-smart-doorlock.src/samsung-smart-doorlock.groovy`

the second directory needs to match the `namespace` attribute or there will be an error when we attempt to deploy

You can either move this from the `smartthings` directory to a `samsungsds` directory or change the `namespace` attribute to `smartthings`

Reply:
@vangoran Can you check this for me? Since I did not upload the source to GitHub, I am not sure about how to handle this. Thanks.

Reply:
@kyunyongpark Okay, I have pushed a patch to move DTH to another directory(devicetypes/samsungsds/samsung-smart-doorlock.src/)